pkg/cephfs/volume.go

```diff
@@ -46,111 +36,27 @@ func getVolumeNamespace(volID volumeID) string {
 	return namespacePrefix + string(volID)
 }
 
-func setVolumeAttribute(root, attrName, attrValue string) error {
-	return execCommandErr("setfattr", "-n", attrName, "-v", attrValue, root)
-}
-
-func createVolume(volOptions *volumeOptions, adminCr *credentials, volID volumeID, bytesQuota int64) error {
-	if err := mountCephRoot(volID, volOptions, adminCr); err != nil {
-		return err
-	}
-	defer unmountCephRoot(volID)
-
-	var (
-		volRoot         = getCephRootVolumePathLocal(volID)
-		volRootCreating = volRoot + "-creating"
-	)
-
-	if pathExists(volRoot) {
-		klog.V(4).Infof("cephfs: volume %s already exists, skipping creation", volID)
-		return nil
-	}
-
-	if err := createMountPoint(volRootCreating); err != nil {
-		return err
-	}
-
-	if bytesQuota > 0 {
-		if err := setVolumeAttribute(volRootCreating, "ceph.quota.max_bytes", fmt.Sprintf("%d", bytesQuota)); err != nil {
-			return err
-		}
-	}
-
-	if err := setVolumeAttribute(volRootCreating, "ceph.dir.layout.pool", volOptions.Pool); err != nil {
```

Comment:
Did we just loose the option to set the data pool for this subvolume? I do not see the `Pool` parameter being passed to create, and neither do we have code here to set the volume attributes, so is this something that is missing and will be handled by the `subvolume` command, or should be done in the CSI code?

Reply:
Yes, subvolume create command doesn't give the option to set the data pool layout. The default data pool of the volume/filesystem is picked. Do we really need the 'data pool' to be configurable?

Reply:
So this is an interesting question, tagging @gman0 as well for his thoughts/requirements.

Currently, we want to make the `pool` specification **optional** (as noted in #380), but not entirely remove it.

For use cases where data needs to reside in a different pool or different type of pool, I would assume this is a useful option to retain. Would this be a correct assumption?

Possibly when multiple FS volumes are supported in the future, we can remove this option, in lieu of creating FS volumes that have differing data pool properties?

Reply:
@batrick ^^

Reply:
Retaining the option to set a data pool seems useful. It also seems fairly trivial to support. Is there a reason we didn't port that over @ajarr?

Reply:
I missed implementing this,
http://tracker.ceph.com/issues/40431

I'm on it.

Reply:
Will add it once the fs command supports the pool arg.

Reply:
@poornimag , with latest nautilus you can set the pool layout, by add the args, `--pool_layoyut <pool_name>``.  can you add this now?